lib/main.gi

```diff
@@ -2713,3 +2713,32 @@ end);
 ##
 #E
 ##
+
+IsMaximal:=function(N,A,D)
+The function IsMaximal checks whether A/D is maximal abelian subgroup of N/D 
+
+----------------------------
+
+IsMaximal:=function(N,A,D)
+local X0, X, Y0, Y, Z, x, W, y;
+X0:=RightTransversal(N,D);
+X:=Filtered(X0,x->not x in A);
+Y0:=RightTransversal(A,D);
+Y:=Filtered(Y0,x->not x in D);
+Z:=[];                                
+for x in X do
+ W:=[];                               
+ for y in Y do
+    if not Comm(x,y) in D then Add(W,y);

```

Comment:
Code formatting style (here and in several more places below): use either 
```
if not Comm(x,y) in D then Add(W,y); fi;
```
or
```
if not Comm(x,y) in D then 
  Add(W,y);
fi;
```